eigenda/certificate.go

```diff
@@ -0,0 +1,175 @@
+package eigenda
+
+import (
+	"fmt"
+	"math/big"
+
+	eigenda_common "github.com/Layr-Labs/eigenda/api/grpc/common"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/Layr-Labs/eigenda/api/grpc/disperser"
+
+	cv_binding "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDACertVerifier"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// EigenDAV1Cert is an internal representation of the encoded cert commitment (i.e, disperser.BlobInfo)
+// read from EigenDA proxy. It is used for type compatibility with the Solidity V1 certificate.
+// This object is encoded into to txs submitted to the SequencerInbox.
+type EigenDAV1Cert struct {
+	BlobVerificationProof cv_binding.BlobVerificationProof `json:"blobVerificationProof"`
+	BlobHeader            cv_binding.BlobHeader            `json:"blobHeader"`
+}
+
+/*
+Unlike 4844 there's no need to inject a version byte into the 0th offset of the hash.
+Taking the hash of commitment + length is key to ensure no trust assumption on the data length
+for one-step proving.
```

Comment:
might be good to add a lot more detail here if this has not been documented elsewhere. It's still unclear to me how this PreimageHash is used, and why using commitment+length is good enough, and it feels like its prob one of the more important things

Reply:
This should be well captured in our existing fork diff documentation